latch_cli/snakemake/single_task_snakemake.py

```diff
@@ -291,5 +293,57 @@ def get_wildcards(self, requested_output, wildcards_dict=None):
 
 RRule.get_wildcards = get_wildcards
 
+
+def get_docker_cmd(
+    img_path,
+    cmd,
+    args="",
+    quiet=False,
+    envvars=None,
+    shell_executable=None,
+    container_workdir=None,
+    is_python_script=False,
+):
+    if shell_executable is None:
+        shell_executable = "sh"
+    else:
+        # Ensure to just use the name of the executable, not a path,
+        # because we cannot be sure where it is located in the container.
+        shell_executable = os.path.split(shell_executable)[-1]
+
+    docker_cmd = "docker run"
+    if envvars:
+        docker_cmd += " " + " ".join(f"-e {k}={v}" for k, v in envvars.items())
+
+    workdir = container_workdir or "/latch"
+    docker_cmd += " -w {}".format(workdir)
+
+    # support for containers currently only works with relative input and output paths
+    # this preserves the behavior before the monkey path
+    docker_cmd += " -v .:{}".format(workdir)
```

Comment:
1. ```suggestion
    docker_cmd += f" --volume .:{workdir}"
```

use fstrings, long-form param names

2. also consider switching to `--mount` as it's more generic

`--mount type=bind,src=.,dst={workdir}`

3. also you need to somehow escape `workdir` - it can't contain `:` or `,` depending on if you use `--volume` or `--mount`

4. also consider mounting read-only? or using an overlay? seems weird to give it read-write to cwd but maybe that's intentional

Reply:
sg, im not mounting as read-only because we want the outputs that are written inside the container to be written to the cwd so that we can upload those files to ldata